src/Avalonia.Base/Input/PointerGesture.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Text;
+using Avalonia.VisualTree;
+
+namespace Avalonia.Input;
+
+public class PointerGesture : IEquatable<PointerGesture>
+{
+    public PointerGesture(MouseButton button, KeyModifiers modifiers = KeyModifiers.None)
+    {
+        Button = button;
+        KeyModifiers = modifiers;
+    }
+
+    public bool Equals(PointerGesture? other)
+    {
+        if (ReferenceEquals(null, other)) return false;
+        if (ReferenceEquals(this, other)) return true;
+
+        return Button == other.Button && KeyModifiers == other.KeyModifiers;
+    }
+
+    public override bool Equals(object? obj)
+    {
+        if (ReferenceEquals(null, obj)) return false;
+        if (ReferenceEquals(this, obj)) return true;
+
+        return obj is KeyGesture gesture && Equals(gesture);
+    }
+
+    public override int GetHashCode()
+    {
+        unchecked
+        {
+            return ((int)Button * 397) ^ (int)KeyModifiers;
+        }
+    }
+
+    public static bool operator ==(PointerGesture? left, PointerGesture? right)
```

Comment:
This class can be a simple record from how it looks.

Reply:
done